cmd/api/main.go

```diff
@@ -169,7 +170,50 @@ func createBook(w http.ResponseWriter, r *http.Request) {
 
 /* Returns a list of the stored books. */
 func getBooks(w http.ResponseWriter, r *http.Request) {
-	returnedBooks, err := listBooks()
+
+	err := r.ParseForm() //Verify errors in request query params and populates r.Form with its values.
+	if err != nil {
+		log.Println(err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	//Adapt query params:
+	name := r.FormValue("name")
+	if name == "" {
+		name = "%"
```

Comment:
This is leaking SQL logic into the request handling layer.
Its better to pass the empty string to the database function and let it decide what that should be in the SQL statement.